apis/v1alpha2/gateway_types.go

```diff
@@ -462,38 +462,44 @@ type GatewayAddress struct {
 	Value string `json:"value"`
 }
 
-// AddressType defines how a network address is represented as a text string.
+// GatewayAddressType defines how a network address is represented as a text
+// string, specifically in the context of listener addresses for Gateways and
+// includes options over the base AddressType type such as Hostname.
 //
 // If the requested address is unsupported, the controller
 // should raise the "Detached" listener status condition on
 // the Gateway with the "UnsupportedAddress" reason.
 //
 // +kubebuilder:validation:Enum=IPAddress;Hostname;NamedAddress
-type AddressType string
+type GatewayAddressType string
```

Comment:
The more I think about this, the more I think that we should share the single AddressType for both use cases and move the enum to the field. Unfortunately field based validation is currently the only way to add experimental additions to enum validation (https://github.com/kubernetes-sigs/gateway-api/blob/master/apis/v1alpha2/httproute_types.go#L498). That won't actually be needed here yet, but if we add a new type in the future, it would be helpful to have it already split out that way.

Reply:
I couldn't find another way to do this, so I implemented this as you suggested. This did not result in a change to anything else during generation so I think we're looking good :+1:

Reply:
1245aab692077a7c130730bf0e30121af9e850d8